gotestcov

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+set -eu
+
+# find_go_mod walks up the directory tree looking for the go.mod file.
+# If it doesn't find it, the script will be aborted.
+find_go_mod() {
+    cwd="$(pwd)"
+
+    while [ "$cwd" != "/" ]; do
+        if [ -f "$cwd/go.mod" ]; then
+            echo "$cwd"
+            return
+        fi
+        cwd=$(dirname "$cwd")
+    done
+    echo "Error: go.mod not found in parent path. Aborting!"
+    exit 1
+}
+
+projectroot="$(find_go_mod)"
+cov_dir="${projectroot}/coverage"
+mkdir -p "${cov_dir}"
+
+# start http server on 6061 if none
+if ! $(nc -z localhost 6061); then
+    nohup python3 -m http.server --directory "${cov_dir}" 6061 1>/dev/null 2>&1 &
+fi
+
+raw_cov_dir="${cov_dir}/raw_files"
+
+rm -fr "${raw_cov_dir}"
+mkdir -p "${raw_cov_dir}"
+
+# Run the tests adding the necessary flags to enable coverage
+go test -cover -covermode=set $@ -args -test.gocoverdir="${raw_cov_dir}"
```

Comment:
Do you mind adding a comment about the non documented flag `-test.gocoverdir`? You can probably just look back at our MM chat when I referenced some links around it to you.